plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java

```diff
@@ -0,0 +1,298 @@
+public class MavenAppEngineStandardWizardPage extends WizardPage implements IWizardPage {
+
+  private String defaultVersion = "0.1.0-SNAPSHOT";
+
+  private Button useDefaults;
+  private Text locationField;
+  private Button locationBrowseButton;
+  private Text groupIdField;
+  private Text artifactIdField;
+  private Text versionField;
+  private Text javaPackageField;
+  private Text projectIdField;
+  
+  public MavenAppEngineStandardWizardPage() {
+    super("basicNewProjectPage"); //$NON-NLS-1$
+    setTitle("Maven-based App Engine Standard Project");
+    setDescription("Create new Maven-based App Engine Standard Project");
+    
+    // todo get a UI designer to pick a better icon (the little plane?) and 
+    // add it to this plugin's icons folder
+    ImageDescriptor descriptor = AbstractUIPlugin.imageDescriptorFromPlugin(
+        "com.google.cloud.tools.eclipse.appengine.localserver", "icons/gcp-32x32.png"); //$NON-NLS-1$ //$NON-NLS-2$
+    this.setImageDescriptor(descriptor);
+
+    setPageComplete(false);
+  }
+
+  @Override
+  public void createControl(Composite parent) {
+    Composite container = new Composite(parent, SWT.NONE);
+    GridLayoutFactory.swtDefaults().numColumns(2).applyTo(container);
+    ModifyListener pageValidator = new PageValidator();
+    
+    /**** Location ****/
+    Group locationGroup = new Group(container, SWT.NONE);
+    locationGroup.setText("Location");
+    GridDataFactory.fillDefaults().span(2, 1).applyTo(locationGroup);
+    GridLayoutFactory.swtDefaults().numColumns(3).applyTo(locationGroup);
+
+    useDefaults = new Button(locationGroup, SWT.CHECK);
+    GridDataFactory.defaultsFor(useDefaults).span(3, 1).applyTo(useDefaults);
+    useDefaults.setText("Create project in workspace");
+    useDefaults.setSelection(true);
+
+    Label locationLabel = new Label(locationGroup, SWT.NONE);
+    locationLabel.setText("Location:");
+    locationLabel
+        .setToolTipText("This location will contain the directory created for the project");
+
+    locationField = new Text(locationGroup, SWT.BORDER);
+    GridDataFactory.swtDefaults().align(SWT.FILL, SWT.CENTER).grab(true, false)
+        .applyTo(locationField);
+    locationField.addModifyListener(pageValidator);
+    locationField.setEnabled(false);
+
+    locationBrowseButton = new Button(locationGroup, SWT.PUSH);
+    locationBrowseButton.setText("Browse");
+    locationBrowseButton.setEnabled(false);
+    locationBrowseButton.addSelectionListener(new SelectionAdapter() {
+      @Override
+      public void widgetSelected(SelectionEvent e) {
+        openLocationDialog();
+      }
+    });
+    useDefaults.addSelectionListener(new SelectionAdapter() {
+      @Override
+      public void widgetSelected(SelectionEvent e) {
+        locationField.setEnabled(!useDefaults.getSelection());
+        locationBrowseButton.setEnabled(!useDefaults.getSelection());
+        checkPageComplete();
+      }
+    });
+    
+
+    /**** Maven Coordinates ****/
```

Comment:
For clarity, can we break this long method up into helper methods, one for each group or section? Or would too much state need to be passed around?